File: HRMS/HRMS.Web/Modules/Communication/SocialPlatformConfig/SocialPlatformConfigColumns.cs
using Serenity.ComponentModel;

namespace HRMS.Communication.Columns;

[ColumnsScript("Communication.SocialPlatformConfig")]
[BasedOnRow(typeof(SocialPlatformConfigRow), CheckNames = true)]
public class SocialPlatformConfigColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int SocialPlatformConfigId { get; set; }
    [EditLink]
    public SocialPlatform Platform { get; set; }
    public string ClientId { get; set; }
    public string RedirectUri { get; set; }
    public bool IsEnabled { get; set; }
}

File: HRMS/HRMS.Web/Modules/ServerTypes/Communication/SocialPlatformConfigColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { SocialPlatform } from "./SocialPlatform";
import { SocialPlatformConfigRow } from "./SocialPlatformConfigRow";

export interface SocialPlatformConfigColumns {
    SocialPlatformConfigId: Column<SocialPlatformConfigRow>;
    Platform: Column<SocialPlatformConfigRow>;
    ClientId: Column<SocialPlatformConfigRow>;
    RedirectUri: Column<SocialPlatformConfigRow>;
    IsEnabled: Column<SocialPlatformConfigRow>;
}

export class SocialPlatformConfigColumns extends ColumnsBase<SocialPlatformConfigRow> {
    static readonly columnsKey = 'Communication.SocialPlatformConfig';
    static readonly Fields = fieldsProxy<SocialPlatformConfigColumns>();
}

[SocialPlatform]; // referenced types